### PR TITLE
[master] Add coverage

### PR DESCRIPTION
**Pull request by atodorov**

### CODE REVIEW

**Comment by atodorov** [comment]
NOTE: This is probably going to collide with #248 once it gets merged!

File: setup.py [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+        _install_data.run(self)
+
+        for old in self.get_outputs():
+            if old.find('/usr/') > -1 and old.endswith('.py'):
+                new = os.path.splitext(old)[0]
+                # move_file doesn't have a way to tell it to overwrite, so remove
+                # the existing copy to accomplish the same thing.
+                if os.path.isfile(new) and os.path.exists(new):
+                    os.unlink(new)
+                move_file(old, new)
```
Comment (atodorov) [inline setup.py]: @bcl this piece of code is supposed to remove the `.py` suffix when installing. I did limited testing but it looks like it works.

**Comment by atodorov** [comment]
@bcl wrt branching: what is the strategy, merge lorax-composer into master at some later point ?

**Comment by bcl** [comment]
> @bcl wrt branching: what is the strategy, merge lorax-composer into master at some later point ?

No. It is based on rhel7-branch and if everything goes right will be merged back into there.

**Comment by atodorov** [comment]
@bcl rebased to lorax-composer branch. Is it OK to merge?

**Comment by atodorov** [comment]
OTOH tests & coverage is something that should really go into master and then be cherry-picked if applicable to the lorax-composer branch. What do you say ?

**Comment by atodorov** [comment]
After discussion with @bcl we've decided to drop the commits which add .py the program names in favor of moving classes and functions out of these files into library modules. This can be done separately when needed so this PR can be merged now.

File: Makefile [hunk 2]
```diff
@@ -25,6 +25,14 @@ check:
 	@echo "*** Running pylint ***"
 	PYTHONPATH=$(PYTHONPATH):./src/ ./tests/pylint/runpylint.py
 
+test:
+	@echo "*** Running tests ***"
+	PYTHONPATH=$(PYTHONPATH):./src/ $(PYTHON) -m nose -v --with-coverage --cover-erase --cover-branches \
+                        --cover-package=pylorax --cover-package=bin --cover-package=sbin --cover-inclusive
+	coverage3 report -m
+	[ -f "/usr/bin/coveralls" ] && [ -n "$(COVERALLS_REPO_TOKEN)" ] && coveralls || echo
```
Comment (bcl) [inline Makefile]: Same question here as on lorax-composer branch, although I'm not as concerned about using 3rd party stuff for Fedora builds. Where does coveralls come from

Comment (atodorov) [reply Makefile]: That's python3-coveralls-1.1-4.fc26.noarch

**Pull request merged**